pkgs/by-name/sp/spades/package.nix

```
@@ -2,33 +2,36 @@
   lib,
   stdenv,
   fetchurl,
-  zlib,
+  zlib-ng,
   bzip2,
   cmake,
   python3,
+  readline,
+  ncurses
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "spades";
-  version = "3.15.5";
+  version = "4.0.0";
 
   src = fetchurl {
     url = "https://github.com/ablab/spades/releases/download/v${finalAttrs.version}/SPAdes-${finalAttrs.version}.tar.gz";
-    hash = "sha256-FVw2QNVx8uexmgUDHR/Q0ZvYLfeF04hw+5O9JBsSu/o=";
+    hash = "sha256-B8AusdnZD2EaxzvdMN3CQu1RsAyKN1cYnoqBN62M+4s=";
   };
   sourceRoot = "SPAdes-${finalAttrs.version}/src";
 
-  env.CXXFLAGS = toString [
-    # GCC 13: error: 'uint32_t' does not name a type
-    "-include cstdint"
-  ];
+  patches = [ ./copytree.patch ];
+
+  cmakeFlags = [ "-DZLIB_ENABLE_TESTS=OFF" ];
 
   nativeBuildInputs = [ cmake ];
 
   buildInputs = [
-    zlib
+    zlib-ng
```

```suggestion
    zlib-ng
```
`z` is last letter of alphabet.

---

Do you mean that we should order the list? Well, it's done. I also reformatted long lines.

---

Yes, whenever possible (not Python, there order matters), it's good to order alphabetically. Looks nicer also.